omnigibson/robots/manipulation_robot.py

```diff
@@ -1260,7 +1262,17 @@ def _load_state(self, state):
         if self.grasping_mode == "physical":
             return
 
-        # TODO: Include AG_state
+        # Include AG_state
+        # TODO: currently doese not take care of cloth objects
+        self._ag_obj_constraint_params = state["ag_obj_constraint_params"]
+        for arm in self._ag_obj_constraint_params.keys():
+            if len(self._ag_obj_constraint_params[arm]) > 0:
+                data = self._ag_obj_constraint_params[arm]
+                obj = og.sim.scene.object_registry("prim_path", data["ag_obj_prim_path"])
+                link = obj.links[data["ag_link_prim_path"].split("/")[-1]]
+                self._ag_data[arm] = (obj, link)
+                self._ag_obj_in_hand[arm] = obj
```

Comment:
Is this line needed? I think it might be handled by `self._establish_grasp` internally.

Reply:
Thank you for pointing this out! The `self._establish_grasp` function sets `self._ag_obj_constraint_params` and `self._ag_obj_in_hand`, so I will remove those. `self._ag_obj_in_hand` does need to be set here since nothing else is updating it when the state is loaded. How does this look?
```
        for arm in state["ag_obj_constraint_params"].keys():
            if len(state["ag_obj_constraint_params"][arm]) > 0:
                data = state["ag_obj_constraint_params"][arm]
                obj = og.sim.scene.object_registry("prim_path", data["ag_obj_prim_path"])
                link = obj.links[data["ag_link_prim_path"].split("/")[-1]]
                self._ag_data[arm] = (obj, link)
                self._establish_grasp(arm=arm, ag_data=self._ag_data[arm], contact_pos=data["contact_pos"])
```

Reply:
looks great!